File: bench/dotnet/base64.cs
namespace Bench;
static class Base64
{
    // Generate N bytes, base64 encode+decode. Checksum = (sum enc char codes +
    // sum decoded bytes) mod 2^31.
    public static void Run(int n)
    {
        long x = 123456789;
        var b = new byte[n];
        for (int i = 0; i < n; i++) { x = (x * 1103515245 + 12345) & 0x7FFFFFFF; b[i] = (byte)(x % 256); }
        string enc = Convert.ToBase64String(b);
        byte[] dec = Convert.FromBase64String(enc);
        long encSum = 0;
        foreach (char c in enc) encSum = (encSum + c) % 2147483647;
        long decSum = 0;
        foreach (byte bb in dec) decSum = (decSum + bb) % 2147483647;
        Console.WriteLine((encSum + decSum) % 2147483647);
    }
}

File: bench/ruby/base64.rb
# Generate N bytes, base64 encode+decode. Checksum = (sum enc char codes + sum
# decoded bytes) mod 2^31.
require "base64"
n = (ENV["BENCH_N"] || "50000").to_i
x = 123456789
bytes = Array.new(n)
n.times do |i|
  x = (x * 1103515245 + 12345) & 0x7FFFFFFF
  bytes[i] = x % 256
end
enc = Base64.strict_encode64(bytes.pack("C*"))
dec = Base64.strict_decode64(enc)
enc_sum = 0
enc.each_byte { |c| enc_sum = (enc_sum + c) % 2147483647 }
dec_sum = 0
dec.each_byte { |c| dec_sum = (dec_sum + c) % 2147483647 }
puts((enc_sum + dec_sum) % 2147483647)
